Framework/Beamline/src/ComponentInfo.cpp

```diff
@@ -139,8 +154,26 @@ ComponentInfo::componentsInSubtree(const size_t componentIndex) const {
   return indices;
 }
 
+const std::vector<size_t> &
+ComponentInfo::children(const size_t componentIndex) const {
+  static const std::vector<size_t> emptyVec;
+
+  if (!isDetector(componentIndex))
+    return (*m_children)[compOffsetIndex(componentIndex)];
+
+  return emptyVec;
+}
+
 size_t ComponentInfo::size() const { return m_size; }
 
+size_t
+ComponentInfo::numberOfDetectorsInSubtree(const size_t componentIndex) const {
+  if (isDetector(componentIndex))
+    return 0;
```

Comment:
This is inconsistent, the return of this method should be them same as `detectorsInSubtree().size()`. That being said, isn't it sufficient to use that to avoid bloating the interface?

Reply:
`detectorsInSubtree()` results in an allocation. I do not want the indices. Just the size. I will look into it.